index.ts

```diff
@@ -1090,22 +1101,85 @@ function asyncSyscallImpl(db: DatabaseFake) {
           ),
         );
       }
+      case "1.0/runUdf": {
+        const {
+          udfType,
+          name,
+          reference,
+          functionHandle,
+          args: udfArgs,
+        } = args;
+        const functionPath = await getFunctionPathFromAddress({
+          name,
+          reference,
+          functionHandle,
+        });
+        if (udfType === "query") {
+          return JSON.stringify(
+            convexToJson(await withAuth().queryFromPath(functionPath, true, udfArgs)),
+          );
+        }
+        if (udfType === "mutation") {
+          return JSON.stringify(
+            convexToJson(
+              await withAuth().mutationFromPath(functionPath, true, udfArgs),
+            ),
+          );
+        }
+        if (udfType === "action") {
+          return JSON.stringify(
+            convexToJson(
+              await withAuth().actionFromPath(functionPath, udfArgs),
+            ),
+          );
+        }
+        throw new Error(
+          `\`convexTest\` does not support udf type: "${udfType}"`,
+        );
+      }
+      case "1.0/createFunctionHandle": {
+        const {
+          name,
+          reference,
+          functionHandle,
+        } = args;
+        const functionPath = await getFunctionPathFromAddress({
+          name,
+          reference,
+          functionHandle,
+        });
+        const handle = `function://${functionPath.componentPath};${functionPath.udfPath}`;
```

Comment:
Can we add a comment about the format we're using here (or maybe even add two methods next to each other for creating + parsing function handle?)